appveyor.yml - fix build error, add trunk

MSP-Greg opened the pull request:
Hoe 3.18.0 won't work with Ruby 2.4, and causes false positives on CI, as the tests DO NOT RUN.  It has been fixed in master.

Fixed other Appveyor issue, added trunk.

hsbt commented on appveyor.yml:
```diff
@@ -10,27 +10,34 @@ install:
-      } else {
-        gem install minitest hoe rake-compiler rubocop        --no-document --conservative
+      if ($rv -eq '_trunk') {
+        $trunk_uri = 'https://ci.appveyor.com/api/projects/MSP-Greg/ruby-loco/artifacts/ruby_trunk.7z'
```
Unfortunately, We never use the 3rd party package without platformer. Because it's hard to know its configuration.

MSP-Greg replied:
Ok.  I guess it's caused a lot of problems in rdoc, reline, & RubyGems (and other external repos)?  Also, who exactly is 'we'?

JFYI:

1. The recent issue in Appveyor (https://ci.appveyor.com/project/ruby/racc/builds/25815248), I think I fixed it, but that's a UI setting that who knows whether Appveyor may reset.  I set my fork UI setting that way, and it passed.

2. If you look here:
https://ci.appveyor.com/project/ruby/racc/builds/25807470/job/tveyvfy72pc7es1s#L269
and
https://travis-ci.org/ruby/racc/jobs/555543268#L500
you'll see that Hoe clobbers Ruby 2.4 rake test.  I locked the Hoe version in Appveyor.yml, whether it should be added to travis, your call.  As mentioned above, it's fixed in Hoe master.

hsbt replied:
You should separate the issues. 

1. Add trunk task. I requested the remove your custom package for trunk.
2. The issues of Hoe

I'm okay to apply your patch for 2.

MSP-Greg commented:
I guess it passed.   I did something for free, I fixed two issues, and I added more testing.  And it only involved CI.

First thing you hit me with is your rules.  As I asked, who is 'we'?

hsbt commented:
> First thing you hit me with is your rules. As I asked, who is 'we'?

The maintainers of ruby organization.

hsbt commented:
Unfortunately, It's hard to maintain for me.
